src/layers/domain/core/questionnaire.py

```diff
@@ -1,75 +1,149 @@
-from datetime import datetime
-from enum import Enum
-from uuid import UUID
+from datetime import date, datetime, time
+from types import FunctionType
+from typing import Generic, Type, TypeVar
 
-from pydantic import BaseModel, Field
+from domain.core.error import DuplicateError, InvalidResponseError
+from domain.core.validation import ENTITY_NAME_REGEX
+from pydantic import BaseModel, Field, validator
 
-from .error import DuplicateError
-from .validation import ENTITY_NAME_REGEX
+# from domain.core.questionnaire_validation_custom_rules import url
 
-
-class QuestionType(Enum):
-    """
-    The data type of the question
-    """
-
-    STRING = str
-    INT = int
-    BOOL = bool
-    DATE_TIME = datetime
+T = TypeVar("T")
+A = {str, int, bool, datetime, float, date, time}  # allowed question types
```

Comment:
tiny one - but I'd renamed this to `ALLOWED_QUESTION_TYPES` and then don't need the comment